spec/openapi.yaml

```diff
@@ -1415,6 +1418,172 @@ paths:
+          - ```sender_address``` is case insensitive. When one already exists for this account, you get a error message ```conflict```;
+          - ```destination_countries``` currently we only support ```AU``` and ```GB``` as the destination country;
+          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```;
+          - ```reason``` for requesting a sender address is required;
+          - ```label``` name of the sender address.
+      operationId: requestSenderAddressUsingPOST
+      parameters: []
+      requestBody:
+        description: ''
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RequestSenderAddress'
+            example:
+              sender_address: EXAMPLE
+              sender_address_type: ALPHANUMERIC
+              usage_type: ALPHANUMERIC
+              destination_countries:
+                - AU
+              reason: 'This is my reason'
+              label: label
+        required: true
+      responses:
+        '201':
+          description: Created
+          headers: {}
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SenderAddressRequestItem'
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '409':
+          description: Conflict
+  /v1/messaging/numbers/sender_address/requests/{id}:
+    get:
+      tags:
+        - Source Address
+      summary: Get status of a sender address request
+      description: |
+        Retrieve the current status of a sender address request using the request ID returned in the sender address request end point.
+        A successful request to the get message status endpoint will return a response body as follows:
+        ```json
+        {
+            "id": "id",
+            "sender_address": "sample",
+            "sender_address_type": "ALPHANUMERIC",
+            "usage_type": "ALPHANUMERIC",
+            "destination_countries": [
+              "AU"
+            ],
+            "reason": "This is my approval reason",
+            "label": "label"
+            "status": "APPROVED",
+            "account_id": "sample",
+            "created_date": "2023-09-07T05:48:26.741Z",
+            "last_modified_date": "2023-09-07T05:49:20.888Z"
+        }
+        ```
+      operationId: GetStatusOfSenderAddressRequest
+      parameters:
+        - name: id
+          in: path
+          description: '36 character UUID.'
+          required: true
+          style: simple
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Get the status of Sender Address Request
+          headers: {}
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SenderAddressRequestItem'
+        '403':
```

Comment:
Need 401 Unauthorized when auth is invalid

Reply:
Done.